test/ibmq/test_ibmq_job.py

```diff
@@ -296,13 +297,55 @@ def test_retrieve_job_error(self, provider):
     def test_retrieve_jobs_status(self, provider):
         """Test retrieving jobs filtered by status."""
         backend = provider.get_backend('ibmq_qasm_simulator')
-        job_list = provider.backends.jobs(backend_name=backend.name(),
-                                          limit=5, skip=0, status=JobStatus.DONE)
-
-        self.assertTrue(job_list)
-        for job in job_list:
+        # Get the most recent jobs that are done.
+        backend_jobs = backend.jobs(limit=10, skip=0, status=JobStatus.DONE)
+        backend_jobs_status_filtered = backend.jobs(limit=10, skip=0, status=[JobStatus.DONE])
+        backend_jobs_str_filtered = backend.jobs(limit=10, skip=0, status=['DONE'])
+        self.assertTrue(backend_jobs)
+        self.assertTrue(backend_jobs_status_filtered)
+        self.assertTrue(backend_jobs_str_filtered)
+        self.assertEqual(backend_jobs, backend_jobs_status_filtered)
+        self.assertEqual(backend_jobs, backend_jobs_str_filtered)
+
+        for job in backend_jobs:
             self.assertTrue(job.status() is JobStatus.DONE)
 
+    @requires_provider
+    def test_retrieve_all_job_statuses(self, provider):
+        """Test retrieving jobs filtered by all possible job statuses."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+        # pylint: disable=unnecessary-comprehension
+        all_job_statuses = [status for status in JobStatus]
+        # Get the most recent jobs with any job status.
+        backend_jobs_filtered = backend.jobs(limit=10, status=all_job_statuses)
+        backend_jobs = backend.jobs(limit=10)
+        self.assertTrue(backend_jobs)
+        self.assertTrue(backend_jobs_filtered)
+        self.assertEqual(backend_jobs, backend_jobs_filtered)
+
+    @requires_provider
+    def test_retrieve_multiple_job_statuses(self, provider):
+        """Test retrieving jobs filtered by multiple job statuses."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+        # Get the most recent jobs that failed or were cancelled.
+        job_statuses_to_filter = [JobStatus.ERROR, JobStatus.CANCELLED]
+        backend_jobs_filtered = backend.jobs(limit=10, status=job_statuses_to_filter)
+        self.assertTrue(backend_jobs_filtered)
+
+        for job in backend_jobs_filtered:
+            self.assertTrue(job.status() in job_statuses_to_filter)
+
+    @requires_provider
+    def test_retrieve_active_jobs(self, provider):
+        """Test retrieving jobs that are currently unfinished."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+        active_job_states = [status for status in ApiJobStatus
+                             if status not in API_JOB_FINAL_STATES]
+        active_jobs = backend.active_jobs()
```

Comment:
It might be worth adding another test/subTest that ensures a `QUEUED` job is returned in `active_jobs()` (by submitting it to the busiest device) since with the simulator the jobs are more likely to be in `RUNNING` state.